src/HeatPump.cpp

```diff
@@ -173,20 +173,30 @@ void HeatPump::setPowerSetting(bool setting) {
   wantedSettings.power = lookupByteMapIndex(POWER_MAP, 2, POWER_MAP[setting ? 1 : 0]) > -1 ? POWER_MAP[setting ? 1 : 0] : POWER_MAP[0];
 }
 
-String HeatPump::getPowerSetting() {
+const char* HeatPump::getPowerSetting() {
   return currentSettings.power;
 }
 
-void HeatPump::setPowerSetting(String setting) {
-  wantedSettings.power = lookupByteMapIndex(POWER_MAP, 2, setting) > -1 ? setting : POWER_MAP[0];
+void HeatPump::setPowerSetting(const char* setting) {
+  int index = lookupByteMapIndex(POWER_MAP, 2, setting);
+  if (index > -1) {
+    wantedSettings.power = POWER_MAP[index];
+  } else {
+    wantedSettings.power = POWER_MAP[0];
+  }
```

Comment:
@SwiCago you wondered why I changed this from the ternary form `a ? b : c` to the `if {} else {}` form... I'd originally intended to add some more code into the `else` clause, but then didn't go ahead with it, but left the change. Happy to put it back if you prefer the ternary form :smile:

Reply:
Whoops, no, the above isn't correct, sorry. I use the pointer from `*_MAP` rather than the one passed in to avoid any memory management fights. If someone were to pass in a string buffer that was heap-allocated, then unallocate it, things might get messy. This way we only store references to data the library itself owns.